pytorch_lightning/trainer/training_loop.py

```diff
@@ -856,6 +862,39 @@ def backward(self, result, optimizer, opt_idx, *args, **kwargs):
             # track gradients
             self.track_and_norm_grad(optimizer=optimizer)
 
+    def should_return_on_invalid_result(self, result):
+        is_result_none = result is None
+        if isinstance(self.trainer.model, LightningDistributedDataParallel):
+            model_ref = self.trainer.get_model()
+            skip_on_at_leat_one = model_ref.decision_on_invalid_result == DecisionOnInvalidResult.SKIP_ON_AT_lEAST_ONE
+            never_skip = model_ref.decision_on_invalid_result == DecisionOnInvalidResult.NEVER_SKIP
+
+            if never_skip and is_result_none:
+                raise MisconfigurationException(
+                    "decision_on_invalid_result ``never_skip`` doesn't support returning None for training_step. "
+                    "Hint: Either switch to ``skip_on_at_least_one`` or return a Nan or Inf loss directly"
```

Comment:
I would do this for the user instead of raising an Exception

Reply:
I would prefer to leave a MisConfigurationException here.

This would be triggered only when people change the property `invalid_loss_strategy` to return `never_skip`. I think they should be aware of what is happening and change their code accordingly.
I don't think Lightning should choose between the 2 modes.

Reply:
I'm talking about `return a Nan or Inf loss directly`